tools/topology/topology1/platform/intel/intel-boards.m4

```diff
@@ -0,0 +1,34 @@
+#
+# Machine driver differentiation for dai link ID
+#
+
+# variable that need to be defined in upper m4
+ifdef(`PLATFORM',`',`fatal_error(note: Need to define platform idendifier for intel-boards)')
+ifdef(`BOARD',`',`fatal_error(note: Need to define board idendifier for intel-boards)')
+
+define(`NUM_DMIC_BE_LINK', `2')
+
+ifelse(
+	PLATFORM, `cml', `define(`NUM_HDMI_BE_LINK', 3)', 
+	PLATFORM, `jsl', `define(`NUM_HDMI_BE_LINK', 3)', 
+	PLATFORM, `tgl', `define(`NUM_HDMI_BE_LINK', 4)', 
+	PLATFORM, `adl', `define(`NUM_HDMI_BE_LINK', 4)',
+	`fatal_error(note: Unknown platform to intel-boards)')
+
+# define the ID base for backend DAI Links
+ifelse(
+	BOARD, `sof_rt5682', `
```

Comment:
It's not really a BOARD, it's a machine driver that you are referring to.

LINUX_MACHINE_DRIVER would be non-ambiguous.

Reply:
fixed